### DISCUSSION TITLE
BootKeyboard

### DISCUSSION THREAD

**Discussion by ktchiam** [NicoHood/HID#290] [Q&A]
Hi,

I'm trying to use Boot Keyboard to send ALT + B keystroke to access BIOS during startup. It seems that the CPU receives SHIFT + ALT + B instead.
Is there anyway I can further check on this?

My code:
```c
#include "HID-Project.h"

const int pinLed = LED_BUILTIN;
const int pinBios = 2;
const int pinPW = 3;

void setup() {
  pinMode(pinLed, OUTPUT);
  pinMode(pinBios, INPUT_PULLUP);
  pinMode(pinPW, INPUT_PULLUP);

  // Sends a clean report to the host. This is important on any Arduino type.
  BootKeyboard.begin();
}


void loop() {
/*  // Light led if keyboard uses the boot protocol (normally while in bios)

  if (BootKeyboard.getProtocol() == HID_BOOT_PROTOCOL)
    digitalWrite(pinLed, HIGH);
  else
    digitalWrite(pinLed, LOW);

*/
  // Trigger ALT B to enter bios
  if (!digitalRead(pinBios)) {
    digitalWrite(pinLed, HIGH);
    BootKeyboard.press(KEY_LEFT_ALT);
    BootKeyboard.press('B');
    delay(100); 
    digitalWrite(pinLed, LOW);
    
    // Simple debounce
    delay(300);
  }

  // Enter bios password
  if (!digitalRead(pinPW)) {
    digitalWrite(pinLed, HIGH);
    BootKeyboard.println("password"); 
    digitalWrite(pinLed, LOW);
    
    // Simple debounce
    delay(300);
  }
}
```

**Comment by NicoHood** [comment]
How do you think it receives a shift signal?

You are only pressing the keys, but never releasing them. That might be also an issue.

**Comment by ktchiam** [comment]
Hi Nico,

Thanks for the reply. I have found the issue to be using capital letter B.
In order to have the capital letter, shift button is also pressed. My
problem resolved after I change to KEY_B.

Yes, you also pointed out a mistake I made.

Thank you again.

Best regards
KT Chiam

On Thu, 3 Jun 2021 at 16:56, NicoHood ***@***.***> wrote:

> How do you think it receives a shift signal?
>
> You are only pressing the keys, but never releasing them. That might be
> also an issue.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/NicoHood/HID/discussions/290#discussioncomment-819650>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AO2IRX2IXD4ANAQDALBTFSTTQ47VJANCNFSM4555EAGQ>
> .
>


-- 
Best Regards,
KT Chiam

**Reply by NicoHood** [reply]
Nice! Very cool it was so easy to solve. I also did not see the B vs b.
